spec/helpers/link_to_helper_spec.rb

```diff
@@ -146,4 +146,32 @@
 
   end
 
+  describe "when constructing a google analytics event string" do
+    it "returns correctly formatted event javascript" do
+      expect(analytics_event_code("test category", "test")).to eq(
+        "if (pageTracker) { pageTracker._trackEvent('test category','test') };"
+      )
+    end
+
+    context "when passed a label value" do
+      it "includes the default value 1 if none is supplied" do
+        expect(analytics_event_code("test category", "test", "test label")).to eq(
+          "if (pageTracker) { pageTracker._trackEvent('test category','test','test label',1) };"
+        )
+      end
+
+      it "uses the supplied value if there is one" do
+        expect(analytics_event_code("test category", "test", "test label", 42)).to eq(
+          "if (pageTracker) { pageTracker._trackEvent('test category','test','test label',42) };"
+        )
+      end
+
+      it "doesn't wrap the label in quotes if passed label_is_script=true" do
+        expect(analytics_event_code("test category", "test", "window.location.href", 42, true)).to eq(
+          "if (pageTracker) { pageTracker._trackEvent('test category','test',window.location.href,42) };"
```

Comment:
The use case for this is the widget code - which does something like `window.top.location.href` which tells us where the code's been embedded, not essential but a nice to have. If we wrap it as a String, it will literally log "window.top.location.href" which isn't what we want to capture (I have accidentally proved this a couple of times).

So yes, depending on the scenario, if we want to use the optional label/value feature we can supply a we can supply a String literal if that's useful or some javascript so that the browser assembles the value that the `onclick` event uses instead.

(or to put it another way... I agree, I need to write some helpful documentation!)